src/freertos_drivers/esp_idf/EspIdfWiFi.hxx

```diff
@@ -1083,32 +1127,61 @@ private:
         nvs_close(cfg);
     }
 
-    /// Find a WiFi STA profile that matches the given SSID
-    /// @param ssid SSID to match
-    /// @param pass where to fill in the password, ignored if nullptr
-    /// @param sec where to fill in the security, ignored if nullptr
-    /// @param index index to start the linear search from
-    /// @return index within the wifi profiles config, else -1 if not found
-    int find_sta_profile(const std::string &ssid, std::string *pass = nullptr,
-        uint8_t *sec = nullptr, uint8_t index = 0) override
+    /// Reset configuration to factory defaults.
+    void factory_default()
     {
-        OSMutexLock locker(&lock_);
-        for (unsigned i = index; i < HWDefs::MAX_STA_PROFILES; ++i)
+        LOG(INFO, "wifi: factory_default()");
+        OSMutexLock locker(&this->lock_);
+
+        // Clear all configuration.
+        memset(&this->userCfg_, 0, sizeof(this->userCfg_));
+
+        // AP setup.
+        WiFiDefs::SecurityType sec = WiFiDefs::SEC_WPA2;
+        if (HWDefs::DEFAULT_AP_PASSWORD[0] == '\0')
         {
-            if (userCfg_.sta_[i].ssid_ == ssid)
-            {
-                if (pass)
-                {
-                    pass->assign(userCfg_.sta_[i].pass_);
-                }
-                if (sec)
-                {
-                    *sec = userCfg_.sta_[i].sec_;
-                }
-                return i;
-            }
+            sec = WiFiDefs::SEC_OPEN;
         }
-        return -1;
+        str_populate<this->MAX_SSID_SIZE>(
```

Comment:
it looks like this code is very similar to the init_config_user implementation in the base class

Reply:
Fixed.